### DISCUSSION TITLE
Stuck on loop where 'Retrying in 10 seconds' after running into errors and omitting frames

### DISCUSSION THREAD

**Discussion by jengalbert2022** [wgu-opensource/osmt#304] [Asking for Help] [answered]
I've ensured that I have the latest release of OSMT cloned, valid Okta credentials, the most up-to-date version of Docker, and a clean Docker container every time I run the ./osmt_cli.sh -q command, after having run the initialization command and changing the file. However, the terminal script gets stuck running loops where it retries in 10 seconds after running into a loop and omitting frames. The log is attached.
[osmtrunlog.txt](https://github.com/wgu-opensource/osmt/files/10690117/osmtrunlog.txt)

**Comment by JohnKallies** [comment, marked as answer]
Looking at your log output for the Spring application, it is looking for an ElasticSearch host named `osmt_quickstart_elasticsearch_1` (with an underscore) on the Docker network. If you look at your docker-compose up output, it's providing a host named `osmt_quickstart-elasticsearch-1` (with a hyphen).

Net result is that your local Docker service is using a hyphen instead of an underscore when naming your services. You should modify your osmt-quickstart.env file, and ensure that those host names match the output of a `docker ps` command.

Please reply back if you need additional clarification?

**Reply by zaReyna** [reply]
I am also facing the same issues. Where can I find the ElasticSearch host name on my local Docker service to match with the one in my osmt-quickstart.env file? 
It also says that my Okta Issuer URL is missing and to add the 'okta.oauth2.issuer' property to the application environments. I already added my Okta Issuer and replaced the values on the osmt-quickstart.env file as per instructions from the README.md file. 

I have attached the log here. [Error .txt](https://github.com/wgu-opensource/osmt/files/10962577/Error.txt)
Thanks in advance for your help.

*Sorry if there is an easy solution to this. I am an intern at my company who is still learning about Docker and interested in using OSMT, which is why I am trying to set up the quickstart.

**Reply by JohnKallies** [reply]
@RinaSotelo Thanks for asking your question.

Run a `docker ps` from your shell. It will be a long line, so make your window large enough to read it cleanly.

Look for the ElasticSearch service, and update the `ELASTICSEARCH_URI=http://osmt_quickstart_elasticsearch_1:9200` in your quickstart env file.

Please reply back if you have trouble?

**Reply by zaReyna** [reply]
@JohnKallies Thank you for your help.
I updated the ElasticSearch in my quickstart env file but now, its not connecting to the localhost (http://localhost:8080/). 
[Error .txt](https://github.com/wgu-opensource/osmt/files/10994956/Error.txt)

I'm still getting an error in the output of my Spring application about not finding my Okta Issuer URL and to add the 'okta.oauth2.issuer' property to the application environments. I tried accessing my local Spring app to see if I could could add the property but its saying that it cannot run OSMT Spring app with Maven [SpringError.txt](https://github.com/wgu-opensource/osmt/files/10994937/SpringError.txt).

**Reply by JohnKallies** [reply]
Ok, thank you.

Please share the output of `./osmt_cli.sh -v`, and please start the Quickstart, and share the output of `docker ps`.

**Reply by zaReyna** [reply]
Here is the output of docker ps: [docker ps.txt](https://github.com/wgu-opensource/osmt/files/10995328/docker.ps.txt)
Here is the output of ./osmt_cli.sh -v: [osmt -v.txt](https://github.com/wgu-opensource/osmt/files/10995331/osmt.-v.txt)

**Reply by JohnKallies** [reply]
@RinaSotelo Thank you for sharing. Here's some feedback about your log output...
- You don't need Maven to run OSMT or a Spring app, but the `osmt_cl.sh` tooling assumes Maven. This is meant for local / development use. Look at the output of `./osmt_cli.sh -v` for dependencies that are not satisfied.
- The URLs for your docker-compose services (MySQL / ElasticSearch / Redis) must match the service names reported in `docker ps`. You need to change the `ELASTICSEARCH_URI` and `REDIS_URI` values in your `osmt-quickstart.env` to match the Docker containers started by your system. Look for difference between hyphens and underscores.
- 
Please reply back if you are still having trouble?

**Comment by JohnKallies** [comment]
@jengalbert2022 I'm just checking in... it looks like you've tried using the OSMT quickstart at different times over the last year. How is it going? What are you / your organization trying to accomplish by this?

Was your issue above resolved by updating service names in the docker-compose file?

**Reply by jengalbert2022** [reply]
Hi! Sorry, I've just jumped back into this recently and the above method does work around the retrying 10 seconds loop. I've been running into other issues. I have two issues I'm running into. The first is something about an OKTA issuer ID. I don't know if they wasn't an actually important  as it goes on and gets stuck on something else. Below is the first (but both will be present in the log)

"Your Okta Issuer URL is missing. You can copy your domain from the Okta Developer Console. Follow these instructions to find it: https://bit.ly/finding-okta-domain
osmt_quickstart-app-1            | To fix this add the `okta.oauth2.issuer` property to your application environments."

The second issue is once it goes past this after a very brief pause and gets stuck on the last 4 lines (I'd assume it would continue generating the same line but its stuck for around 15 minutes before I give up and just copy the log).
[log4_5_23_2_59pm.txt](https://github.com/wgu-opensource/osmt/files/11162324/log4_5_23_2_59pm.txt)
[log4_5_23_3_21pm.txt](https://github.com/wgu-opensource/osmt/files/11162331/log4_5_23_3_21pm.txt)

At the end of both log it gets stuck on the same process/node

**Reply by jengalbert2022** [reply]
And here's another log when it ran for a lot longer (closer to 40 min)
[log4_5_23_4_29pm.txt](https://github.com/wgu-opensource/osmt/files/11162845/log4_5_23_4_29pm.txt)

**Reply by jengalbert2022** [reply]
Also I'm part of evrmore (I have another email registered with OSMT as albert@evrmore.io)

**Reply by JohnKallies** [reply]
For what it's worth, I ran a QuickStart off the develop branch today, and it worked fine. It took about 10 minutes to start.

Some follow up points...
- Please ensure you are on the latest from the `develop` branch
- Compare your modified `osmt-quickstart.env` to the `osmt-quickstart.env.example` in the source code. Ensure that your env vars are correct, and double check the values from within your Okta web application (see the section about Okta in the README.md`)
- You never said what happens when you open a browser to http://localhost:8080. What happens?
- Please ensure you have suitable RAM for your Docker service. Locally, I have 8GB and can run 2 OSMT instances side-by-side.

**Reply by JohnKallies** [reply]
Also, thank you for sharing other communication paths. WGU is not in the professional service market around OSMT, and the best we can offer is Discussion forum support here. I hope we can get you running with this.

**Reply by jengalbert2022** [reply]
When I launch http://localhost:8080 it just says server not found. I checked in the two env files is the example file has '-' where i have '_' (to match the docker) under the elasticsearch_uri and redis_uri. I have 32 GB ram on the MacBook Pro 14
